recipes/duckdb/all/conanfile.py

```diff
@@ -0,0 +1,197 @@
+        tc.variables["DUCKDB_DEV_ITERATION"] = 0
+        tc.variables["BUILD_ICU_EXTENSION"] = self.options.with_icu
+        tc.variables["BUILD_PARQUET_EXTENSION"] = self.options.with_parquet
+        tc.variables["BUILD_TPCH_EXTENSION"] = self.options.with_tpch
+        tc.variables["BUILD_TPCDS_EXTENSION"] = self.options.with_tpcds
+        tc.variables["BUILD_FTS_EXTENSION"] = self.options.with_fts
+        tc.variables["BUILD_HTTPFS_EXTENSION"] = self.options.with_httpfs
+        tc.variables["BUILD_VISUALIZER_EXTENSION"] = self.options.with_visualizer
+        tc.variables["BUILD_JSON_EXTENSION"] = self.options.with_json
+        tc.variables["BUILD_EXCEL_EXTENSION"] = self.options.with_excel
+        tc.variables["BUILD_SQLSMITH_EXTENSION"] = self.options.with_sqlsmith
+        tc.variables["BUILD_ODBC_DRIVER"] = self.options.with_odbc
+        tc.variables["FORCE_QUERY_LOG"] = self.options.with_query_log
+        tc.variables["BUILD_SHELL"] = self.options.with_shell
+        tc.variables["DISABLE_THREADS"] = not self.options.with_threads
+        tc.variables["BUILD_UNITTESTS"] = False
+        tc.variables["BUILD_RDTSC"] = self.options.with_rdtsc
+        tc.variables["EXTENSION_STATIC_BUILD"] = not self.options.shared
+        tc.variables["ENABLE_SANITIZER"] = False
+        tc.variables["ENABLE_UBSAN"] = False
+        if is_msvc(self) and not self.options.shared:
+            tc.preprocessor_definitions["DUCKDB_API"] = ""
+        tc.generate()
+
+        dpes = CMakeDeps(self)
+        dpes.generate()
+
+    def build(self):
+        apply_conandata_patches(self)
+        if is_msvc(self) and not self.options.shared:
+            replace_in_file(self, os.path.join(self.source_folder, "src", "include", "duckdb.h"),
+                "#define DUCKDB_API __declspec(dllimport)",
+                "#define DUCKDB_API"
+            )
+            replace_in_file(self, os.path.join(self.source_folder, "src", "include", "duckdb", "common", "winapi.hpp"),
+                "#define DUCKDB_API __declspec(dllimport)",
+                "#define DUCKDB_API"
+            )
+
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
+        cmake.install()
+
+        if self.options.shared:
+            rm(self, "*.a", os.path.join(self.package_folder, "lib"))
+            rm(self, "duckdb_*.lib", os.path.join(self.package_folder, "lib"))
+
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "cmake"))
+
+    def package_info(self):
+        if self.options.shared:
+            self.cpp_info.libs = ["duckdb"]
+        else:
+            self.cpp_info.libs = [
+                "duckdb_static",
+                "duckdb_fmt",
```

Comment:
Does this vendor packagesm

Reply:
Yes.
These packages have been modified to have duckdb own namespaces. 
like this : https://github.com/duckdb/duckdb/blob/master/third_party/fmt/include/fmt/core.h#L163-L166

Reply:
Ughs that's like death for us lol okay.